workflow-core/src/commonMain/kotlin/com/squareup/workflow1/BaseRenderContext.kt

```diff
@@ -127,221 +126,250 @@ public interface BaseRenderContext<out PropsT, StateT, in OutputT> {
+ *        }
+ *      }
+ *    )
+ *
+ *  with this:
+ *
+ *    return SomeScreen(
+ *      onClick = context.eventHandler("onClick") { state = SomeNewState }
+ *    )
+ *
+ * Notice how your [update] function is passed to the [actionSink][BaseRenderContext.actionSink]
+ * to be eventually executed as the body of a [WorkflowAction]. If several actions get stacked
+ * up at once (think about accidental rapid taps on a button), that could take a while.
+ *
+ * If you require something to happen the instant a UI action happens, [eventHandler]
+ * is the wrong choice. You'll want to write your own call to `actionSink.send`:
+ *
+ *    return SomeScreen(
+ *      onClick = {
+ *        // This happens immediately.
+ *        MyAnalytics.log("SomeScreen was clicked")
+ *
+ *        context.actionSink.send(
+ *          action("onClick") {
+ *            // This happens eventually.
+ *            state = SomeNewState
+ *          }
+ *        }
+ *      }
+ *    )
+ *
+ * @param name A string describing the update, included in the action's [toString]
+ * as a debugging aid
+ * @param update Function that defines the workflow update.
+ */
+public fun <PropsT, StateT, OutputT> BaseRenderContext<PropsT, StateT, OutputT>.eventHandler(
+  name: String,
+  update: WorkflowAction<
+    PropsT,
+    StateT,
+    OutputT
+    >.Updater.() -> Unit
+): () -> Unit {
+  val handler = {
+    actionSink.send(action("eH:$name", update))
   }
+  return remember(name) { handler }
+}
 
-  public fun <EventT> eventHandler(
-    name: String,
-    update: WorkflowAction<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT>.Updater.(
-      EventT
-    ) -> Unit
-  ): (EventT) -> Unit {
-    return { event ->
-      actionSink.send(action("eH:$name") { update(event) })
-    }
+public inline fun <reified EventT, PropsT, StateT, OutputT>
+  BaseRenderContext<PropsT, StateT, OutputT>.eventHandler1(
+  name: String,
+  crossinline update: WorkflowAction<PropsT, StateT, OutputT>.Updater.(
+    EventT
+  ) -> Unit
+): (EventT) -> Unit {
+  val handler: (EventT) -> Unit = { event ->
+    actionSink.send(action("eH:$name") { update(event) })
   }
+  val key = "$name+${EventT::class.simpleName}"
+  return remember(key) { handler }
+}
 
-  public fun <E1, E2> eventHandler(
-    name: String,
-    update: WorkflowAction<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT>.Updater.(
-      E1,
-      E2
-    ) -> Unit
-  ): (E1, E2) -> Unit {
-    return { e1, e2 ->
-      actionSink.send(action("eH:$name") { update(e1, e2) })
-    }
+public inline fun <reified E1, reified E2, PropsT, StateT, OutputT>
+  BaseRenderContext<PropsT, StateT, OutputT>.eventHandler2(
+  name: String,
+  crossinline update: WorkflowAction<PropsT, StateT, OutputT>.Updater.(
+    E1,
+    E2
+  ) -> Unit
+): (E1, E2) -> Unit {
+  val handler: (E1, E2) -> Unit = { e1, e2 ->
+    actionSink.send(action("eH:$name") { update(e1, e2) })
   }
+  val key = "$name+${E1::class.simpleName}+${E2::class.simpleName}"
```

Comment:
Using `simpleName` here b/c KMP & Javascript.

- [ ] Find a way to use `expect` / `actual` to use `qualifiedName` when it's available